src/ascii_str.rs

```diff
@@ -576,6 +576,43 @@ pub trait AsMutAsciiStr {
     fn as_mut_ascii_str(&mut self) -> Result<&mut AsciiStr, AsAsciiStrError>;
 }
 
+// These generic implementations mirror the generic implementations for AsRef<T> in core.
+impl<'a, T> AsAsciiStr for &'a T where T: AsAsciiStr + ?Sized {
```

Comment:
I'll change this and the following trait implementations to
```
impl<'a, T: ?Sized> AsAsciiStr for &'a T where T: AsAsciiStr {
```
even though it's not as nicely formatted as your version to keep compatibility with rust 1.16